tests/dragonfly/utility.py

```diff
@@ -15,68 +20,462 @@ def grouper(n, iterable):
+
+        time_start = time.time()
+
+        cmdcount = await producer
+        for consumer in consumers:
+            await consumer
+
+        took = time.time() - time_start
+        qps = round(cmdcount * self.dbcount / took, 2)
+        print(f"Filling took: {took}, QPS: {qps}")
+
+    def stop(self):
+        """Stop all invocations to run"""
+        self.stop_flag = True
+
+    def reset(self):
+        """ Reset internal state. Needs to be called after flush or restart"""
+        self.gen.reset()
+
+    async def capture(self, port=None, target_db=0, keys=None):
+        """Create DataCapture for selected db"""
+        if port is None:
+            port = self.port
+
+        if keys is None:
+            keys = sorted(list(self.gen.keys_and_types()))
+
+        client = aioredis.Redis(port=port, db=target_db)
+        return DataCapture(await self._capture_entries(client, keys))
+
+    async def compare(self, initial_capture, port=6379):
+        """Compare data capture with all dbs of instance and return True if all dbs are correct"""
+        print(f"comparing capture to {port}")
+        keys = sorted(list(self.gen.keys_and_types()))
+        captures = await asyncio.gather(*(
+            self.capture(port=port, target_db=db, keys=keys) for db in range(self.dbcount)
+        ))
+        for db, capture in zip(range(self.dbcount), captures):
+            if not initial_capture.compare(capture):
+                eprint(f">>> Inconsistent data on port {port}, db {db}")
+                return False
+        return True
+
+    def target(self, key_cnt):
+        self.gen.key_cnt_target = key_cnt
+
+    async def _generator_task(self, queues, target_times=None, target_deviation=None):
+        cpu_time = 0
+        submitted = 0
+        deviation = 0.0
+
+        file = None
+        if self.log_file:
+            file = open(self.log_file, 'a')
+
+        def should_run():
+            if self.stop_flag:
+                return False
+            if target_times is not None and submitted >= target_times:
+                return False
+            if target_deviation is not None and abs(1-deviation) < target_deviation:
+                return False
+            return True
+
+        while should_run():
+            start_time = time.time()
+            blob, deviation = self.gen.generate()
+            cpu_time += (time.time() - start_time)
+
+            await asyncio.gather(*(q.put(blob) for q in queues))
+            submitted += 1
+
+            if file is not None:
+                file.write('\n'.join(blob))
+
+            print('.', end='', flush=True)
+            await asyncio.sleep(0.0)
+
+        print("\ncpu time", cpu_time, "batches", submitted)
+
+        await asyncio.gather(*(q.put(None) for q in queues))
+        for q in queues:
+            await q.join()
+
+        if file is not None:
+            file.flush()
+
+        return submitted * self.gen.batch_size
+
+    async def _consumer_task(self, db, queue):
```

Comment:
what ut does ? its consuming semantics are internal to the seeder, right?
maybe `_executor_task` ?